packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -322,15 +318,25 @@ type RestrictedMenuItemProps = {
   dataFences?: TDataFence[];
   children: ReactNode;
 };
+type TLongLivedFlag = {
+  value: boolean;
+  reason?: string;
+};
 const restrictedMenuItemDefaultProps: Pick<
   RestrictedMenuItemProps,
   'permissions'
 > = {
   permissions: [],
 };
+function isLongLivedFlag(
+  flag: TFlagVariation | TLongLivedFlag
+): flag is TLongLivedFlag {
+  return typeof (flag as TLongLivedFlag)?.value === 'boolean';
+}
 const RestrictedMenuItem = (props: RestrictedMenuItemProps) => {
   // NOTE: Custom application are activated/deactivated while their
   // visibility is not controlled via a visibiility overwrite.
+  const flagVariation = useFlagVariation(props.featureToggle);
```

Comment:
Note, that with `useFlagVariation(undefined)` as `props.featureToggle` not being defined `@flopflip` will return `false` which in turn on L374 we don't use. A culprit to not being able to use hooks conditionally as we know.